client/src/pages/Login.js

```diff
@@ -36,9 +36,73 @@ const Login = () => {
   const classes = useStyles();
   const [email, setEmail] = useState("");
   const [password, setPassword] = useState("");
+  const [error, setError] = useState("");
+  const [emailError, setEmailError] = useState(null);
+  const [passwordError, setPasswordError] = useState(null);
+
+  const checkEmailError = () => {
+    if (emailError) {
+      return emailError;
+    } else return null;
+  };
+
+  const checkPasswordError = () => {
+    if (passwordError) {
+      return passwordError;
+    } else return null;
+  };
+
+  useEffect(() => {
+    if (error.includes("Email")) {
+      setEmailError(error);
+    } else if (error.includes("Password")) {
+      setPasswordError(error);
+    } else {
+      setEmailError(null);
+      setPasswordError(null);
+    }
+  }, [error]);
+
+  // remove errors when user starts typing
+  useEffect(() => {
+    if (emailError) {
+      setEmailError(null);
+    }
+  }, [email]);
+
+  useEffect(() => {
+    if (passwordError) {
+      setPasswordError(null);
+    }
+  }, [password]);
 
   const submit = () => {
-    return; // placeholder
+    async function login(user) {
+      try {
+        let res = await fetch("/login", {
```

Comment:
let's favour const over let. Only use let if you actually need to re-bind a variable

Reply:
got it